OutOfSchool/OutOfSchool.WebApi/Controllers/V1/ApplicationController.cs

```diff
@@ -360,4 +387,22 @@ public async Task<IActionResult> AllowedToReview(Guid parentId, Guid workshopId)
     {
         return Ok(await applicationService.AllowedToReview(parentId, workshopId).ConfigureAwait(false));
     }
+
+    private async Task<bool> IsCurrentUserBlocked()
+    {
+        var userId = GettingUserProperties.GetUserId(User);
+
+        return await userService.IsBlocked(userId);
+    }
+
+    private async Task<bool> IsWorkshopOrProviderBlocked(Guid workshopId)
+    {
+        var isWorkshopBlocked = await workshopService.isBlocked(workshopId).ConfigureAwait(false);
+
+        var providerId = await providerService.GetProviderIdForWorkshopById(workshopId).ConfigureAwait(false);
+
+        var isProviderBlocked = await providerService.IsBlocked(providerId).ConfigureAwait(false);
```

Comment:
(Probably performance) There are 3 calls to services that might be potentially transformed to 3 calls to db. As I can't check now if these calls are cached, I'm asking you to consider related performance issues. 

I do not insist on massive rewriting, just asking if it could be easily fixed, or should be registered as tech debt, or could be ignored.

Reply:
I have performed stricter checks in this case. Made changes. We only need to check workshops because blocking the provider also blocks all his workshops.  Thus, there is no need to check the provider.